tools/scripts/gitrelease.pl

```diff
@@ -286,25 +277,46 @@ sub create_dummy_release_file {
   my $filename = basename($path);
   print "Creating dummy $filename because of mock release\n";
   $dummy_release_files{$filename} = 1;
-  open(my $fh, '>', $path) or die "Couldn't open file $path: $!";
+  open(my $fh, '>', $path) or trace("Couldn't open file $path: $!");
   print $fh "This is a dummy file because the release was mocked\n";
   close($fh);
 }
 
 sub new_pithub {
   my $settings = shift();
+  my %args = @_;
+  my $needs_token = $args{needs_token} // 0;
+  my $user = $args{user};
+  my $repo = $args{repo};
+
+  if (defined($user) || defined($repo)) {
+    # We need to create another Pithub object, make a copy of settings for this.
+    $settings = {%{$settings}};
```

Comment:
I don't know if I've seen this Perl construct before. Is it the same as `$settings = \%{$settings};` (if that even works)?
May also be good to use two different variables.

Reply:
I'm not sure what `$settings = \%{$settings};` is supposed to do, but here I'm dereferencing settings and putting the values in a new hash. I don't think I can make two separate variables because usage of copied settings is conditional.